src/digraph.rs

```diff
@@ -2720,6 +2720,45 @@ impl PyDiGraph {
         self.clone()
     }
 
+    /// Reverse the direction of all edges in the graph, in place.
+    ///
+    /// This method modifies the graph instance to reverse the direction of all edges.
+    /// It does so by iterating over all edges in the graph and removing each edge,
+    /// then adding a new edge in the opposite direction with the same weight.
+    ///
+    ///  .. note:: The edge indices will change if this method is used to reverse the edges
+    ///
+    /// For example::
+    ///
+    ///     import rustworkx as rx
+    ///
+    ///     graph = rx.PyDiGraph()
+    ///     graph.add_nodes_from(range(4))
+    ///     graph.add_edge(0, 1, 3)
+    ///     graph.add_edge(1, 2, 5)
+    ///     graph.add_edge(2, 3, 2)
+    ///
+    ///     graph.reverse_inplace()
+    ///
+    ///     assert graph.edge_list() == [(3, 2), (2, 1), (1, 0)]
+    #[pyo3(text_signature = "(self)")]
+    pub fn reverse_inplace(&mut self, py: Python) {
+        let indices = self.graph.edge_indices().collect::<Vec<EdgeIndex>>();
+        for idx in indices {
+            let (source_node, dest_node) = self
+                .graph
+                .edge_endpoints(idx)
+                .expect("Edge Index received from iterator");
+            let weight = self
+                .graph
+                .edge_weight(idx)
+                .expect("Edge Index received from iterator")
```

Comment:
```suggestion
                .unwrap()
```

If this fails is because of PyO3.  It panics and there is not much point in printing a message

Reply:
yup :( all hell breaks loose